ergotree-ir/src/sigma_protocol/sigma_boolean.rs

```diff
@@ -220,4 +251,16 @@ mod arbitrary {
 }
 
 #[cfg(test)]
-mod tests {}
+mod tests {
+    use super::*;
+    use crate::serialization::sigma_serialize_roundtrip;
+    use proptest::prelude::*;
+
+    proptest! {
+
+        #[test]
+        fn sigma_boolean_ser_roundtrip(v in any::<SigmaBoolean>()) {
+            prop_assert_eq![sigma_serialize_roundtrip(&v), v]
```

Comment:
When I print out the generated SigmaBoolean value on my local machine, I ALWAYS get the following:
ProofOfKnowledge(ProveDlog(ProveDlog { h: EC:0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798 }))

Should this value not be changing as it is randomly generated?

Reply:
Good catch! I've been thinking about that for a while. Although we have `DlogProverInput::random()` from which we can get a random `EcPoint` with `DlogProverInput::public_image()` it's in `ergotree-interpreter` crate. I was thinking of moving it to `ergotree-ir` but eventually declined since it belongs to an "interpreter's" part of the sigma protocol.
So I suggest to make `EcPoint::random()` (exp the group generator) and use it in `Arbitrary` impl for `EcPoint` (see `Arbitrary` impl for `DlogProverInput`).